lib/cubit/security/security_cubit.dart

```diff
@@ -49,17 +49,30 @@ class SecurityCubit extends Cubit<SecurityState> with HydratedMixin<SecurityStat
   }
 
   Future<bool> testPin(String pin) async {
+    // Allow access if PIN is disabled
```

Comment:
Since this method already takes a pin code it is weird that it also treats cases where the pin is not needed.
I think we should make the check on whether pin is needed or not before we show the pin widget to the user, hence before calling this method.
If we call this method then it should be strict with pin checking.